node/p2p_block_sync.md

```diff
@@ -0,0 +1,91 @@
+# P2P Block Sync
+
+## Abstract
+
+P2P Block Sync enables rollkit full nodes, including aggregators, to gossip blocks amongst  
+themselves and have lower rollup block times than the DA layer block times.
+
+```mermaid
+sequenceDiagram
+    title P2P Block Sync
+
+    participant User
+    participant Block Producer
+    participant Full Node 1
+    participant Full Node 2
+    participant DA Layer
+
+    User->>Block Producer: Send Tx
+    Block Producer->>Block Producer: Generate Block
+    Block Producer->>DA Layer: Publish Block
+
+    Block Producer->>Full Node 1: Gossip Block
+    Block Producer->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Verify Block
+    Full Node 1->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Mark Block Soft-Confirmed
+
+    Full Node 2->>Full Node 2: Verify Block
+    Full Node 2->>Full Node 2: Mark Block Soft-Confirmed
+
+    DA Layer->>Full Node 1: Retrieve Block
+    Full Node 1->>Full Node 1: Mark Block Hard-Confirmed
```

Comment:
I just realized that we need to handle the case of the network pruning the data. It could be that we cannot retrieve the data itself from the DA-Layer anymore. So we might want to start saving inclusion proofs that prove that the data was published at some point. Or we only look at the DA-Layer during a certain period. (Fraud Proof Window) . This is actually an open question and should be explored further.

Reply:
Ideally, the DALC handles that case with archival nodes with expired DA state being set up. Also, nodes can decide to prune or not, so as long as there is an altrustic full node that did not prune and is willing to share. It also works then.